### ISSUE TITLE
Stateful RNN mod in sampling

### ISSUE THREAD

**Issue by aliabbasjp** [jcjohnson/torch-rnn#38]
Does just removing the lines:  self:resetStates() in the sample function in langmodel.lua will make it a statefull RNN, thus preserving long term context?

**Comment by AlekzNet** [comment]
I removed these lines, and the loss immediately dropped by 15%. Thanks for the suggestion! But it can be very data-dependent.

**Comment by AlekzNet** [comment]
Interesting, every checkpoint saving resets the state. I wonder if this is the reason why the training loss jumps a bit up after every save:

Epoch 1.24 / 50, i = 20 / 4250, loss = 1.025447	
Checkpoint = 	cv/checkpoint_20.t7	
Epoch 1.25 / 50, i = 21 / 4250, loss = 1.042428	
. . .
Epoch 1.35 / 50, i = 30 / 4250, loss = 1.013393	
Checkpoint = 	cv/checkpoint_30.t7	
Epoch 1.36 / 50, i = 31 / 4250, loss = 1.036500	
. . .
Epoch 1.47 / 50, i = 40 / 4250, loss = 1.003105	
Checkpoint = 	cv/checkpoint_40.t7	
Epoch 1.48 / 50, i = 41 / 4250, loss = 1.020587

**Comment by dgcrouse** [comment]
Over a year since last comment, closing stale issue.
